R/fe_native.R

```diff
@@ -0,0 +1,116 @@
+#' Check species status (native, exotic, ...) for one species from Flora Europaea webpage
+#'
+#' @importFrom XML xmlValue getNodeSet htmlTreeParse
+#' @export
+#'
+#' @param sp character; a vector of length one with a single scientific species names in the form of
+#'    c("Genus species").
+#' @return A list of vectors containing the countries where the species is native, exotic, ...
+#'
+#' @description This function check the status (native or exotic) of a species in each of the eu countries.
+#'
+#' For that end, it checks Flora Europaea (http://rbg-web2.rbge.org.uk/FE/fe.html) and scrps the data from there.
+#'
+#' Note that the webpage contains more information.
+#'
+#' As expected, the function is as good as the database is. I think for native species is robust but
+#' new exotic species are not added as to my knowledge the database is not updated anymore.
+#' The database is not able to recognize species
+#' synonyms.
+#'
+#' See http://rbg-web2.rbge.org.uk/FE/data/countries for explanation of the database codes.
+#'
+#' @author Ignasi Bartomeus \email{nacho.bartomeus@@gmail.com}
+#' @examples \donttest{
+#' sp <- c("Lavandula stoechas", "Carpobrotus edulis", "Rhododendron ponticum",
+#'        "Alkanna lutea", "Anchusa arvensis")
+#' fe_native(sp[1])
+#' sapply(sp, fe_native, simplify = FALSE)
+#' }
+#'
+
+fe_native <- function(sp){
+  #add country short-names translation cheat sheet as dataframe
+  country <- data.frame(short = c("Al", "Au", "Az", "Be", "Bl", "Br", "Bu", "Co", "Cr", "Cz",
+                                  "Da", "Fa", "Fe", "Ga", "Ge", "Gr", "Hb",
+                                  "He", "Ho", "Hs", "Hu", "Is", "It", "Ju",
+                                  "Lu", "No", "Po", "Rm", "Rs", "Sa","Sb",
+                                  "Si", "Su", "Tu", "N", "B", "C",
+                                  "W", "K", "E"),
+                        long = c("Albania", "Austria", "Azores", "Belgium", "Islas_Baleares",
+                                 "Britain", "Bulgaria", "Corse", "Kriti",
+                                 "Czechoslovakia", "Denmark", "Faroer",
+                                 "Finland", "France", "Germany", "Greece",
+                                 "Ireland", "Switzerland", "Netherlands", "Spain",
+                                 "Hungary", "Iceland", "Italy", "Jugoslavia",
+                                 "Portugal", "Norway", "Poland", "Romania",
+                                 "USSR", "Sardegna", "Svalbard", "Sicilia",
+                                 "Sweden", "Turkey", "USSR_Northern_Division",
+                                 "USSR_Baltic_Division", "USSR_Central_Division",
+                                 "USSR_South_western", "USSR_Krym",
+                                 "USSRSouth_eastern_Division"))
+  require(plyr)
+  require(XML)
+  require(RCurl)
```

Comment:
@ibartomeus don't do require/library calls in the function itself. can you remove these 3 lines?  

also, doesn't look like you use plyr here